addons/point_of_sale/models/pos_session.py

```diff
@@ -2076,7 +2076,7 @@ def _loader_params_product_product(self):
                 'fields': [
                     'display_name', 'lst_price', 'standard_price', 'categ_id', 'pos_categ_ids', 'taxes_id', 'barcode',
                     'default_code', 'to_weight', 'uom_id', 'description_sale', 'description', 'product_tmpl_id', 'tracking',
-                    'write_date', 'available_in_pos', 'attribute_line_ids', 'active', 'image_128', 'combo_ids',
+                    'write_date', 'available_in_pos', 'attribute_line_ids', 'active', 'image_128', 'combo_ids', 'name',
```

Comment:
This won't correct the error it will just make it less problematic. The problem here is that you solve a specific case, what could be done is either to handle the error in a better way and telling the user that a certain reward domain will not work in the pos (maybe through a notification or something like that), either checking every domain for the rewards we want to load and if some domains use fields that we do not load, add those fields to the list of field to load. Both solutions can be argued and there is maybe another better solution but I do not think of other for the moment. Don't hesitate to contact me to discuss more about this if you want

Reply:
I think that doing a non breaking error is the best choice as it is difficult to reload the products after you load the rewards to figure out which fields need to be loaded.